src/main/java/org/opentripplanner/updater/vehicle_sharing/vehicles_positions/BikeStationsGraphWriterRunnable.java

```diff
@@ -44,21 +50,46 @@ private boolean addBikeStationToGraph(BikeRentalStation station, Graph graph) {
         return true;
     }
 
-    private boolean updateBikeStationInfo(BikeRentalStation station, Graph graph) {
-        RentBikeEdge rentVehicleEdge = graph.bikeRentalStationsInGraph.getOrDefault(station, null);
-        if (rentVehicleEdge != null) {
-            rentVehicleEdge.getBikeRentalStation().bikesAvailable = station.bikesAvailable;
-            rentVehicleEdge.getBikeRentalStation().spacesAvailable = station.spacesAvailable;
-        } else {
-            return addBikeStationToGraph(station, graph);
-        }
-        return true;
+    private void updateBikeStationInfo(BikeRentalStation station, RentBikeEdge rentBikeEdge) {
+        rentBikeEdge.getBikeRentalStation().bikesAvailable = station.bikesAvailable;
+        rentBikeEdge.getBikeRentalStation().spacesAvailable = station.spacesAvailable;
     }
 
 
     @Override
     public void run(Graph graph) {
-        int count = (int) bikeRentalStationsFetchedFromApi.stream().filter(station -> updateBikeStationInfo(station, graph)).count();
-        LOG.info("Placed {} bike stations on a map", count);
+        int updatedStationsCount = 0;
+        int removedStationsCount;
+        int placedStationsCount = 0;
+        int failedToPlaceCount = 0;
+        Map<BikeRentalStation, RentBikeEdge> oldStations = graph.bikeRentalStationsInGraph;
+        graph.bikeRentalStationsInGraph = new HashMap<>();
+
+        for (BikeRentalStation station : bikeRentalStationsFetchedFromApi) {
```

Comment:
to jest jakoś pokrętnie napisane, ale niech będzie

Reply:
Wydaje mi się, że myślą przewodnią jest
`TemporaryVertex.disposeAll(dissappearedStations);`
więc updatowanie istniejącego mapowania by nie przeszło tak łatwo.
Wydaje się ok.